flake/packages.nix

```
@@ -7,8 +7,15 @@
 {
 
   perSystem =
-    { pkgs, system, ... }:
     {
+      pkgs,
+      system,
+      config,
+      ...
+    }:
+    {
+      checks = config.packages;
```

Sorry for the poor phrasing, but maybe it's worth saying why we do this:

```suggestion
      # `nix flake check` will validate that all packages are derivations, but it'll only build outputs in `checks`.
      # We want to build all `packages`, so add them to the `checks` output.
      checks = config.packages;
```

?